egs/babel_multilang/s5/local/nnet3/run_tdnn_multilingual.sh

```diff
@@ -0,0 +1,296 @@
+      echo "Pooling training data in $multi_dir_data on" `date`
+      echo ---------------------------------------------------------------------
+      mkdir -p $multi_dir_data
+      combine_lang_list=""
+      for lang_index in `seq 0 $[$num_langs-1]`;do
+        combine_lang_list="$combine_lang_list data/${lang_list[$lang_index]}/train${suffix}_hires"
+      done
+      utils/combine_data.sh data/multi/train${suffix}_hires $combine_lang_list
+      utils/validate_data_dir.sh --no-feats data/multi/train${suffix}_hires
+      touch data/multi/train${suffix}_hires/.done
+    fi
+  fi
+  if [ ! -f $global_extractor/extractor/.done ]; then
+    echo "$0: Generate global i-vector extractor on pooled data from all "
+    echo "languages in $multi_data_dir, using an LDA+MLLT transform trained "
+    echo "on ${lang_list[0]}."
+    local/nnet3/run_shared_ivector_extractor.sh  \
+      --suffix $suffix \
+      --stage $stage ${lang_list[0]} \
+      $multi_data_dir $global_extractor || exit 1;
+    touch $global_extractor/extractor/.done
+  fi
+  echo "$0: Extracts ivector for all languages using $global_extractor/extractor."
+  for lang_index in `seq 0 $[$num_langs-1]`; do
+    local/nnet3/extract_ivector_lang.sh --stage $stage \
+      --train-set train$suffix \
+      --ivector-suffix $ivector_suffix \
+      ${lang_list[$lang_index]} \
+      $ivector_extractor || exit;
+  done
+fi
+
+for lang_index in `seq 0 $[$num_langs-1]`; do
+  multi_data_dirs[$lang_index]=data/${lang_list[$lang_index]}/train${suffix}${feat_suffix}
+  multi_egs_dirs[$lang_index]=exp/${lang_list[$lang_index]}/nnet3/egs${ivector_suffix}
+  multi_ali_dirs[$lang_index]=exp/${lang_list[$lang_index]}/${alidir}${suffix}
+  multi_ivector_dirs[$lang_index]=exp/${lang_list[$lang_index]}/nnet3/ivectors_train${suffix}${ivector_suffix}
+done
+
+if $use_ivector; then
+  ivector_dim=$(feat-to-dim scp:${multi_ivector_dirs[0]}/ivector_online.scp -) || exit 1;
+else
+  echo "$0: Not using iVectors in multilingual training."
+  ivector_dim=0
+fi
+feat_dim=`feat-to-dim scp:${multi_data_dirs[0]}/feats.scp -`
+
+if [ $stage -le 9 ]; then
+  echo "$0: creating multilingual neural net configs using the xconfig parser";
+  if [ -z $bnf_dim ]; then
+    bnf_dim=1024
+  fi
+  input_layer_dim=$[3*$feat_dim+$ivector_dim]
+  mkdir -p $dir/configs
+  cat <<EOF > $dir/configs/network.xconfig
+  input dim=$ivector_dim name=ivector
+  input dim=$feat_dim name=input
+  output name=output-tmp input=Append(-1,0,1,ReplaceIndex(ivector, t, 0))
+
+  # please note that it is important to have input layer with the name=input
+  # as the layer immediately preceding the fixed-affine-layer to enable
+  # the use of short notation for the descriptor
+  # the first splicing is moved before the lda layer, so no splicing here
+  relu-renorm-layer name=tdnn1 input=Append(input@-2,input@-1,input,input@1,input@2,ReplaceIndex(ivector, t, 0)) dim=$input_layer_dim
```

Comment:
The $input_layer_dim (set earlier to 3*$feat_dim+$ivector_dim) here is not compatible to the Append here (which does splicing of 5 frames). Either increase the $input_layer_dim or reduce the splicing here. Do you even need the extra 'dim' expression here?

(I'm following this PR from time-to-time)